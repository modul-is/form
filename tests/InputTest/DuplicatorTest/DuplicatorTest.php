<?php

declare(strict_types=1);

namespace ModulIS\Form\Tests;

require_once __DIR__ . '/../../bootstrap.php';

use Tester\Assert;

class DuplicatorTest extends TestCase
{
	public function testRender()
	{
		$form = $this->getForm();

		$duplicator = $form->addDuplicator('duplicator', function(\ModulIS\Form\DuplicatorContainer $container)
		{
			$container->addText('text', 'text');

			$container->addSubmit('del', 'Smazat')
				->setValidationScope(null)
				->addRemoveOnClick();
		});

		$duplicator->addSubmit('add', 'PÅ™idat')
			->setValidationScope(null)
			->addCreateOnClick(true);

		$duplicator->setValues([['text' => 'Text']]);

		$html = str_replace(["\t", "\n", "\r"], '', file_get_contents(__DIR__ . '/basic.latte'));

		Assert::same($html, $form->getComponent('duplicator')->render()->__toString());
	}


	public function testRenderCustomTemplate()
	{
		$form = $this->getForm();

		$duplicator = $form->addDuplicator('duplicator', function(\ModulIS\Form\DuplicatorContainer $container)
		{
			$container->addText('text', 'text');

			$container->addSubmit('del', 'Smazat')
				->setValidationScope(null)
				->addRemoveOnClick();
		});

		$duplicator->addSubmit('add', 'PÅ™idat')
			->setValidationScope(null)
			->addCreateOnClick(true);

		$duplicator->setTemplate(__DIR__ . '/custom.latte');

		$string = 'custom-template';

		Assert::same($string, $form->getComponent('duplicator')->render());
	}


	public function testRenderSkip()
	{
		$form = $this->getForm();

		$duplicator = $form->addDuplicator('duplicator', function(\ModulIS\Form\DuplicatorContainer $container)
		{
			$container->addText('text', 'text');

			$container->addSubmit('del', 'Smazat')
				->setValidationScope(null)
				->addRemoveOnClick();
		});

		$duplicator->addSubmit('add', 'PÅ™idat')
			->setValidationScope(null)
			->addCreateOnClick(true);

		$duplicator->setAutoRenderSkip();

		$string = '';

		Assert::same($string, $form->getComponent('duplicator')->render());
	}
}

$testcase = new DuplicatorTest;
$testcase->run();
