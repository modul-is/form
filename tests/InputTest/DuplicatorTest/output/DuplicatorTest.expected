<div class="mb-3 col-12"><div id="containerDuplicator" class="card card-accent-primary"><div class="card-body"><div class="row"><div class="mb-3 col-12"><div class="row"><div class="align-self-center col-sm-4"><label for="frm-form-duplicator-0-text" class="col-form-label">text</label></div><div class="align-self-center col-sm-8"><div class="input-group"><input type="text" name="duplicator[0][text]" id="frm-form-duplicator-0-text" value="Text" class="form-control"></div></div></div></div><div class="mb-3 col-12"><button class="btn btn-xs btn-outline-danger float-end" name="duplicator[0][del]" formnovalidate="" type="submit"><span class="fal fa-times fa-fw"></span>Smazat</button></div><hr /></div></div><div class="card-footer"><button class="btn btn-outline-primary float-left btn-xs " name="duplicator[add]" value="Přidat" formnovalidate="" data-nette-validation-scope='["multiplier"]' label="Přidat" type="submit"><span class="fal fa-plus fa-fw"></span>Přidat</button></div></div></div>